src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -59,18 +67,37 @@ class GcsActor {
     auto dummy_object = TaskSpecification(task_spec).ActorDummyObject().Binary();
     actor_table_data_.set_actor_creation_dummy_object_id(dummy_object);
 
+    actor_table_data_.set_language(task_spec.language());
+    actor_table_data_.set_extension_data(
+        task_spec.actor_creation_task_spec().extension_data());
+    actor_table_data_.set_max_task_retries(
+        task_spec.actor_creation_task_spec().max_task_retries());
+    actor_table_data_.set_execute_out_of_order(
+        task_spec.actor_creation_task_spec().execute_out_of_order());
+    actor_table_data_.set_max_pending_calls(
+        task_spec.actor_creation_task_spec().max_pending_calls());
+    actor_table_data_.mutable_function_descriptor()->CopyFrom(
+        task_spec.function_descriptor());
+
     actor_table_data_.set_is_detached(actor_creation_task_spec.is_detached());
     actor_table_data_.set_name(actor_creation_task_spec.name());
     actor_table_data_.mutable_owner_address()->CopyFrom(task_spec.caller_address());
 
     actor_table_data_.set_state(rpc::ActorTableData::DEPENDENCIES_UNREADY);
-    actor_table_data_.mutable_task_spec()->CopyFrom(task_spec);
 
     actor_table_data_.mutable_address()->set_raylet_id(NodeID::Nil().Binary());
     actor_table_data_.mutable_address()->set_worker_id(WorkerID::Nil().Binary());
 
     actor_table_data_.set_ray_namespace(ray_namespace);
 
+    task_spec_ = std::make_unique<rpc::TaskSpec>(task_spec);
```

Comment:
Can you clean up the task spec when the actor state becomes DEAD?

Reply:
it's already been cleared in https://github.com/ray-project/ray/blob/master/src/ray/gcs/gcs_server/gcs_actor_manager.cc#L728
but i agree with you it should be more neat. I'd like to optimize them in future PRs.